misk/web/@misk/components/src/components/TopbarComponentDimensionAware.tsx

```diff
@@ -0,0 +1,388 @@
+  switch (environment) {
+    case Environment.DEVELOPMENT:
+      return color.blue
+    case Environment.TESTING:
+      return color.indigo
+    case Environment.STAGING:
+      return color.green
+    case Environment.PRODUCTION:
+      return color.red
+  }
+}
+
+export class TopbarComponentDimensionAware extends React.Component<
+  IDimensionAwareProps & ITopbarProps,
+  {}
+> {
+  public state = {
+    isOpen: false
+  }
+
+  public render() {
+    const { isOpen } = this.state
+    const {
+      environment,
+      environmentBannerVisible,
+      error,
+      homeName,
+      homeUrl,
+      links,
+      width,
+      status
+    } = this.props
+    return (
+      <MiskNavbar>
+        <MiskNavbarButton onClick={this.handleClick}>
+          <MiskNavbarIcon
+            iconSize={32}
+            icon={isOpen ? IconNames.CROSS : IconNames.MENU}
+          />
+        </MiskNavbarButton>
+        <ResponsiveContainer>
+          <MiskNavbarGroup align={Alignment.LEFT} className="bp3-dark">
+            {this.renderNavbarHome(homeName, homeUrl)}
+            {this.NavbarItems(environment)
+              .slice(0, Math.floor(Math.min(width - 300, 1800) / 400))
+              .map(item => item)}
+          </MiskNavbarGroup>
+        </ResponsiveContainer>
+        <MiskCollapse isOpen={isOpen} keepChildrenMounted={true}>
+          <MiskMenu>
+            <ResponsiveContainer>
+              <MiskMenuNavbarItems>
+                <FlexContainer>
+                  {this.NavbarItems(environment).map(item => (
+                    <span key={item.key} onClick={this.handleClick}>
+                      {item}
+                    </span>
+                  ))}
+                </FlexContainer>
+              </MiskMenuNavbarItems>
+              {links ? (
+                this.renderMenuCategories(links)
+              ) : (
+                <ErrorCalloutComponent error={error} />
+              )}
+            </ResponsiveContainer>
+          </MiskMenu>
+        </MiskCollapse>
+        {this.renderBanner(status, environment, environmentBannerVisible)}
+      </MiskNavbar>
+    )
+  }
+
+  private NavbarItems(environment?: Environment) {
+    console.log(this.props.navbarItems)
+    return [this.renderNavbarEnvironment(environment)].concat(
+      this.props.navbarItems
+        ? this.props.navbarItems.map(item => (
+            <MiskNavbarHeading>
+              <TextHTMLOrElementComponent content={item} />
+            </MiskNavbarHeading>
+          ))
+        : []
+    )
+  }
+
+  private renderBanner(
+    status?: string | Element | JSX.Element,
+    environment?: Environment,
+    environmentBannerVisible: Environment[] = [
+      Environment.DEVELOPMENT,
+      Environment.STAGING,
+      Environment.TESTING
+    ]
+  ) {
+    if (environmentBannerVisible.includes(environment)) {
+      if (typeof status === "string" && !status.startsWith("<")) {
+        console.log(`[STATUS] ${status}`)
```

Comment:
What's the convention on console messages in Misk? Is this for debugging?

Reply:
No convention yet but it is on our todo list #422